base_layer/core/src/covenants/macros.rs

```diff
@@ -20,7 +20,16 @@
 //  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
 //  USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-/// Simple syntax for expressing covenants.
+/// This macro has three different patterns that it can match against based on the syntax provided.
+///
+/// The first pattern matches when the macro is called with an identifier followed by parentheses and optional arguments
+/// ($token($($args:tt)*)). This pattern is useful when you want to create a covenant with some specific arguments.
+///
+/// The second pattern matches when the macro is called with just an identifier followed by empty parentheses
+/// ($token()). This pattern is useful when you want to create a covenant without any arguments.
+///
+/// The third pattern matches when the macro is called with empty parentheses (()). This pattern is used when you want
+/// to create a covenant with no additional arguments. Simple syntax for expressing covenants.
```

Comment:
This is really the implementation details of the macro, which probably isn't that useful for a user. We could expand on this by providing details on the syntax.

e.g. @ indicates an argument literal. @uint(x) @field::features::output_type @hash(x) etc anything without an '@' denotes a filter.

Reply:
You called and here I am!